From d05ee872ecb8ceb67e56a0c1651a6612694654c6 Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Mon, 15 Apr 2024 10:48:44 +0200
Subject: [PATCH 0456/1695] scpfw: module: clock: remove const containts in
 module configuration

Removes the const attributes used for driver_id and api_id fields
of struct mod_clock_dev_config. This change allows to build the
SCP firmware clocks configuration data during initialization rather
than using C source tables defined at build time.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git dcb35f7e3c9d43bfdee03145e747235e8d2b5ba8 scpfw: module: clock: remove const containts in module configuration]
Change-Id: I31e59b1dec185b007457cca5c740015f1913518f
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
---
 .../scmi-server/SCP-firmware/module/clock/include/mod_clock.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/core/lib/scmi-server/SCP-firmware/module/clock/include/mod_clock.h b/core/lib/scmi-server/SCP-firmware/module/clock/include/mod_clock.h
index 73554fb45..fd8efbb37 100644
--- a/core/lib/scmi-server/SCP-firmware/module/clock/include/mod_clock.h
+++ b/core/lib/scmi-server/SCP-firmware/module/clock/include/mod_clock.h
@@ -175,10 +175,10 @@ struct mod_clock_config {
  */
 struct mod_clock_dev_config {
     /*! Reference to the device element within the associated driver module */
-    const fwk_id_t driver_id;
+    fwk_id_t driver_id;
 
     /*! Reference to the API provided by the device driver module */
-    const fwk_id_t api_id;
+    fwk_id_t api_id;
 
     /*!
      * \brief Reference to the element or module that is the source of the
-- 
2.47.2

